sdk/core/Azure.Core/Directory.Build.props

```diff
@@ -2,5 +2,10 @@
   <!--
     Add any shared properties you want for the projects under this package directory that need to be set before the auto imported Directory.Build.props
   -->
+
+  <PropertyGroup>
+    <RunCodeChecksWithThisProject>true</RunCodeChecksWithThisProject>
```

Comment:
Not just for Key Vault, but seems all of these should go in sdk/<ServiceDirectory>/Directory.Build.props. Track 1 targets wouldn't use it, so it doesn't hurt anything there. And all track 2 projects will get it. Many - perhaps even most - service directories have multiple track 2 projects.